test/libsolidity/SemanticTest.cpp

```diff
@@ -410,7 +410,8 @@ TestCase::TestResult SemanticTest::runTest(
 
 			test.setFailure(!m_transactionSuccessful);
 			test.setRawBytes(std::move(output));
-			test.setContractABI(m_compiler.contractABI(m_compiler.lastContractName(m_sources.mainSourceFile)));
+			if (test.call().kind != FunctionCall::Kind::LowLevel)
+				test.setContractABI(m_compiler.contractABI(m_compiler.lastContractName(m_sources.mainSourceFile)));
```

Comment:
Looks like the ABI is indeed used only for formatting parameters (in `TestFunctionCall::format()`) and I see no reason why we'd do that for a low-level call where we just specify raw calldata.

Just to be safe, it would probably not hurt to add an assert that the ABI is non-empty just before it's passed to `ContractABIUtils::parametersFromJsonOutputs()`. But I'd expect that function to fail on some assert when the ABI is empty anyway so it's probably already safe even now.

Reply:
From what I saw, it returns a `nullopt` in that case, which I expect to be treated properly by users of `TestFunctionCall::format()`